### PR TITLE
Ensure Continue token is proxied for openshift RBAC list calls

### PR DESCRIPTION
**Pull request by liggitt**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1574404

### CODE REVIEW

File: pkg/authorization/registry/clusterrole/proxy.go [hunk 1]
```diff
@@ -51,15 +51,14 @@ func (s *REST) List(ctx apirequest.Context, options *metainternal.ListOptions) (
 		return nil, err
 	}
 
-	ret := &authorizationapi.ClusterRoleList{}
+	ret := &authorizationapi.ClusterRoleList{ListMeta: roles.ListMeta}
```
Comment (enj) [inline pkg/authorization/registry/clusterrole/proxy.go]:
```go
// ListMeta describes metadata that synthetic resources must have, including lists and
// various status objects. A resource may have only one of {ObjectMeta, ListMeta}.
type ListMeta struct {
	// selfLink is a URL representing this object.
	// Populated by the system.
	// Read-only.
	// +optional
	SelfLink string `json:"selfLink,omitempty" protobuf:"bytes,1,opt,name=selfLink"`

	// String that identifies the server's internal version of this object that
	// can be used by clients to determine when objects have changed.
	// Value must be treated as opaque by clients and passed unmodified back to the server.
	// Populated by the system.
	// Read-only.
	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#concurrency-control-and-consistency
	// +optional
	ResourceVersion string `json:"resourceVersion,omitempty" protobuf:"bytes,2,opt,name=resourceVersion"`

	// continue may be set if the user set a limit on the number of items returned, and indicates that
	// the server has more data available. The value is opaque and may be used to issue another request
	// to the endpoint that served this list to retrieve the next set of available objects. Continuing a
	// list may not be possible if the server configuration has changed or more than a few minutes have
	// passed. The resourceVersion field returned when using this continue value will be identical to
	// the value in the first response.
	Continue string `json:"continue,omitempty" protobuf:"bytes,3,opt,name=continue"`
}
```

Could the `SelfLink` cause issues?  It is used by the table printing logic from what I can see.

**Comment by enj** [comment]
One comment, but going to LGTM since I do not think it will be an issue.

/lgtm

File: pkg/authorization/registry/clusterrole/proxy.go [hunk 2]
```diff
@@ -51,15 +51,14 @@ func (s *REST) List(ctx apirequest.Context, options *metainternal.ListOptions) (
...
 	for _, curr := range roles.Items {
 		role, err := util.ClusterRoleFromRBAC(&curr)
 		if err != nil {
 			return nil, err
 		}
 		ret.Items = append(ret.Items, *role)
 	}
-	ret.ListMeta.ResourceVersion = roles.ResourceVersion
```
Comment (enj) [inline pkg/authorization/registry/clusterrole/proxy.go]: We could a `ret.ListMeta.SelfLink = ""` here if needed.

File: pkg/authorization/registry/clusterrole/proxy.go (hunk 1 above)
Comment (liggitt) [reply pkg/authorization/registry/clusterrole/proxy.go]: no, selflink gets stomped during serialization

**Pull request merged**